main.go

```diff
@@ -289,3 +324,65 @@ func hasLicense(b []byte) bool {
 	return bytes.Contains(bytes.ToLower(b[:n]), []byte("copyright")) ||
 		bytes.Contains(bytes.ToLower(b[:n]), []byte("mozilla public"))
 }
+
+// reLicense contains regexp to parse years in actual header
+// 2020
+// 2018-2020
+var reLicense = regexp.MustCompile(`(([\d]{4})-)?([\d]{4})`)
+
+// Year contains information about parsed header
+type Year struct {
+	creation         string
+	lastModification string
+	currentString    string
+}
+
+// parseLicenseYear returns Year representation of actual license header
+func parseLicenseYear(b []byte) (bool, *Year) {
+	n := 1000
+	if len(b) < 1000 {
+		n = len(b)
+	}
+	years := reLicense.FindAllSubmatch(b[:n], 2)
+	if len(years) > 0 {
+		yearMatch := years[0]
+		y := &Year{
+			currentString:    string(yearMatch[0]),
+			lastModification: string(yearMatch[3]),
+		}
+		if !bytes.Equal(yearMatch[2], []byte("")) {
+			y.creation = string(yearMatch[2])
+		} else {
+			y.creation = y.lastModification
+		}
+		return true, y
+	}
+
+	return false, nil
+}
+
+// isOutdated return true if the year in license header is older than the current one
+func isOutdated(b []byte, currentYear string) bool {
+	found, years := parseLicenseYear(b)
+	if found {
+		return years.lastModification != currentYear
+	}
+	return false
+}
+
+// updateExistingLicense update license with currentYear
+func updateExistingLicense(b []byte, currentYear string) ([]byte, error) {
+	found, years := parseLicenseYear(b)
+	if !found {
+		return []byte{}, errors.New("cannot parse license header")
+	}
+
+	i := bytes.Index(b, []byte(years.currentString))
```

Comment:
I think that this line could be moved after the check for `update is not required`

Reply:
fair point